### PR TITLE
Add an alternative lookup for the method __delete__ for data descriptors

### PR DESCRIPTION
**Pull request by geryogam**
From your (excellent) blog article [*Unravelling attribute access in Python*](https://snarky.ca/unravelling-attribute-access-in-python/):

> There are two kinds of descriptors: data and non-data. Both kind of descriptors define a `__get__` method for getting what the attribute should be. Data descriptors also define `__set__` and `__del__` methods while non-data descriptors do not; `property` is a data descriptor, `classmethod` and `staticmethod` are non-data descriptors.

But the alternative lookup for the method `__delete__` (note also the typo `__del__` in your article) is absent from your code.

### CODE REVIEW

File: desugar/builtins.py [hunk 1]
```diff
@@ -108,13 +111,14 @@ def __getattribute__(self, attr: str, /) -> Any:
 
         if attr in self.__dict__:
             return self.__dict__[attr]
+        elif descriptor_type_get is not NOTHING:
```
Comment (brettcannon) [inline desugar/builtins.py:114]: Why this change?

**Review by brettcannon — CHANGES_REQUESTED** [review]
Unfortunately I need tests to make sure I don't break this and to verify this is how CPython operates.

File: desugar/builtins.py (hunk 1 above)
Comment (maggyero) [reply desugar/builtins.py:114]: Because this makes the code simpler by avoiding unnecessary nested if statements (as checking `type_attr is not NOTHING` is not necessary as it is already implied by `descriptor_type_get is not NOTHING`).

**Comment by brettcannon** [comment]
I actually have a fix for the other issue you reported that subsumes this PR, so I'm going to close it and give you co-author credit on the PR I'm working on.

**Comment by brettcannon** [comment]
Fix in https://github.com/brettcannon/desugar/commit/2661c22263f0b63c2f3ad1630a0195120e23d750.

**Comment by maggyero** [comment]
Nice update, thanks!
